dataprocspawner/spawner.py

```diff
@@ -809,6 +812,13 @@ def _apply_users_configs(self, cluster_data):
       except:
         pass
 
+    if self.user_options.get('custom_labels'):
+      if 'labels' in cluster_data:
```

Comment:
This will always clear all labels from the template cluster if the user sets any labels. Let's allow the user to overwrite or add labels, but keep any existing labels by default. We can do this by just removing the `del cluster_data['labels']` line.

Reply:
Yes, true. I thought, that's how it has to be - user overwrites all labels, defined in YAML. Otherwise, user defined labels are added to existing list. If you suggest to keep existing one, I'll remove `del` line

Reply:
We generally try to not overwrite value when several values are possible (ex: pip packages). Otherwise, you're right, users can overwrite. For labels, we think it makes more sense to add labels.